_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,176 @@
+revision would be considered a version. The revisions prior to first version
+would have been known as the working copy as each one was created.
+
+If the version is then checked out for further changes (which may or may not be
+published) each new revision becomes the working copy.
+
+Sophisticated servers may support multiple versions of a resource as well as
+multiple working copies of a resource (e.g., to support multiple
+languages or within a version control system which supports multiple branches).
+
+This profile creates a standard for JSON:API to support these diverse versioning
+schemes.
+# Query Parameter
+
+## Usage
+
+An endpoint **MAY** support a `resource_version` query parameter to allow a
+client to indicate which version(s) of a resource should be returned.
+
+If an endpoint does not support the `resource_version` parameter, it **MUST**
+respond with `400 Bad Request` to any requests that include it.
+
+If an endpoint supports the `resource_version` parameter and a client supplies
+it:
+
+  - The server’s response **MUST** contain the most appropriate version of the
+    resource requested.
+  - The server **MUST NOT** include a version of a resource inappropriate for the
+    requested version.
+  - The server **MUST** respond with `404 Not Found` if an appropriate version of
+    the resource requested cannot be located. 
+    
+> Note: This means that a server should not provide "fallbacks" unless the
+> behavior is well defined by the version negotiation mechanism (see below).
+> These rules apply to individual and collection endpoints alike.
+
+## Format
+  
+The value of the `resource_version` parameter **MUST** be a colon-separated
+(U+003A COLON, “:”) string. The first segment of the string **SHOULD** be
+interpreted as an identifier for a _version negotiation mechanism_. A version
+negotiation mechanism defines how a server will locate an appropriate resource
+version. Subsequent segments of the string **SHOULD** be interpreted as version
+negotiation arguments for the preceding mechanism. Collectively, this query
+parameter value is known as the _version identifier_.
+
+> Note: For example, a server may support both ID-based and time-based
+> mechanisms for requesting a resource version. The former mechanism would be
+> useful for comparing versions and the latter could be useful for requesting a
+> resource as it existed at an arbitrary point in time. This profile does not
+> attempt to define every possible mechanism for versioning resources.
+
+```
+                   version-identifier
+                   _______|_________
+                  /                \
+?resource_version=rel:latest-version
+                  \_/ \____________/
+                   |        |
+         version-negotiator |
+                     version-argument
+```
+
+## Server Responsibilities
+
+A server **MUST** respond with `400 Bad Request` if a version negotiator is not
```

Comment:
It seems maybe weird to use `400` here and `501` below?

Reply:
I agree. I was trying to preserve the ability to add new relative version arguments later because I haven't fully worked them out, but also make this a general rule rather than a rule specific to the relative version negotiator.

I think that was probably mistake. I'm happy with this: 9d9e2634a0c9fdf89361329d684ac5f0d5569a05